docs/proposals/super-simple-discovery-api.md

```diff
@@ -0,0 +1,205 @@
+
+### Method: Out of Band
+
+The simplest way to do this would be to simply put this object in a file and copy it around.  This is more overhead for the user, but it is easy to implement and lets users rely on existing systems to distribute configuration.
+
+For the `kubeadm` flow, the command line might look like:
+
+```
+kubeadm join --cluster-info-file=my-cluster.yaml
+```
+
+Note that TLS bootstrap (which establishes a way for a client to authenticate itself to the server) is a separate issue and has its own set of methods.  This command line may have a TLS bootstrap token (or config file) on the command line also.
+
+### Method: HTTPS endpoint
+
+If the ClusterInfo information is hosted in a trusted place via HTTPS you can just request it that way.  This will use the root certificates that are installed on the system.  It may or may not be appropriate based on the user's constraints.
+
+```
+kubeadm join --cluster-info-url="https://example/mycluster.yaml"
+```
+
+This is really a shorthand for someone doing something like (assuming we support stdin with `-`):
+
+```
+curl https://example.com/mycluster.json | kubeadm join --cluster-info-file=-
+```
+
+If the user requires some auth to the HTTPS server (to keep the ClusterInfo object private) that can be done in the curl command equivalent.  Or we could eventually add it to `kubeadm` directly.
+
+### Method: Token
+
+There won’t always be a trusted external endpoint to talk to and transmitting
+the locator file out of band is a pain.  However, we want something more secure
+than just hitting HTTP and trusting whatever we get back.  In this case, we
+assume we have the following:
+
+  * An address for at least one of the API servers (which will implement this API).
+    * This address is technically an HTTPS URL base but is often expressed as a bare domain or IP.
+  * A shared secret token
+
+An interesting aspect here is that this information is often easily obtained before the API server is configured or started.  This makes some cluster bring-up scenarios much easier.
+
+The user experience for joining a cluster would be something like:
+
+```
+kubeadm join --token=ae23dc.faddc87f5a5ab458 <address>
+```
+
+**Note:** This is logically a different use of the token from TLS bootstrap.  We harmonize these usages and allow the same token to play double duty.
+
+#### Implementation Flow
+
+`kubeadm` will implement the following flow:
+
+* `kubeadm` connects to the API server address specified over TLS.  As we don't yet have a root certificate to trust, this is an insecure connection and the server certificate is not validated.
+  * Implementation note: the API server doesn't have to expose a new and special insecure HTTP endpoint.
+* `kubeadm` requests a ConfigMap containing the kubeconfig file defined above.
+  * This ConfigMap exists at a well known URL: `https://<server>//api/v1/namespaces/kube-public/configmaps/cluster-info`
```

Comment:
Plan was we fetch it over https and ignore trust issues at this phase. (just to avoid the overhead of explicitly having to setup non-TLS listeners just to not trust them)